### DISCUSSION TITLE
which version of the binary file should I download?

### DISCUSSION THREAD

**Discussion by gdtv** [rustic-rs/rustic#1192] [Q&A] [answered]
I am a beginner; I don't know which version of the binary file I should download

My machine is x86 64bit Debian 12

which should I download ?
rustic-v0.7.0-x86_64-unknown-linux-gnu.tar.gz
rustic-v0.7.0-x86_64-unknown-linux-musl.tar.gz
......

**Comment by intgr** [comment, marked as answer]
**"gnu" is the correct one to use for most Linux distros**, which use GNU libc. This includes Debian.

"musl" libc is only used by very few distros that are optimized for small install sizes, such as Alpine Linux, OpenWrt and postmarketOS.

(One day hopefully distros will start shipping [`cargo binstall` command](https://github.com/cargo-bins/cargo-binstall) ~~and rustic will add support for it~~ -- that ought to take out this guesswork)

**Comment by aawsome** [comment]
The `musl` version also should work where the `gnu` doesn't, e.g. due to incompatible glibc versions. But yes, trying the `gnu` version first is the right thing to do!

@intgr rustic should actually already support `cargo binstall`. This is another way to install it.

**Reply by intgr** [reply]
> rustic should actually already support cargo binstall

Oh, my bad. I tested `cargo binstall rustic` but had forgotten that the crate name is actually `rustic-rs`.

**Reply by aawsome** [reply]
Unfortunately the name `rustic` was already reserved on crates.io - and when we tried to kindly ask the owners if they can transfer it, they soon started to publish some things there...

**Reply by gdtv** [reply]
gnu:

```
rustic -V
rustic: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by rustic)
rustic: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by rustic)
rustic: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by rustic)
rustic: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by rustic)
```

musl:
```
rustic -V
rustic v0.7.0
```

So now I'm using the `musl` version
